hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -978,3 +997,114 @@ def array_for(index_shape, size):
     return result_shape.flatmap(
         lambda index_shape: st.tuples(*(array_for(index_shape, size) for size in shape))
     )
+
+
+def _from_type(thing: Type[Ex]) -> Optional[st.SearchStrategy[Ex]]:
+    """Called by st.from_type to try to infer a strategy for thing using numpy.
+
+    If we can infer a dtype strategy for thing, we return that; otherwise,
+    returns None (or raises).
```

Comment:
I'm not a fan of interfaces that catch _any_ error and move on; it's too easy to miss unexpected errors.  Can we return `None` for all cases we can't infer?  Or if not, restrict the expected error types?

(the answer may well be "it's not worth the effort in this case", e.g. because it's version-dependent)

Reply:
No problem, it is in fact the docstring that is wrong here (the callsite does not consume exceptions)